android/daemon/src/main/java/org/mozilla/firefox/vpn/daemon/GleanUtil.kt

```diff
@@ -34,7 +36,12 @@ class GleanUtil(aParent: Context) {
             "production"
         }
         val conf = Configuration(Configuration.DEFAULT_TELEMETRY_ENDPOINT, channel, 500, HttpURLConnectionUploader())
-        val build = BuildInfo("versionCode", "VersionName", Calendar.getInstance())
+        val info: PackageInfo = mParent.packageManager.getPackageInfo(mParent.packageName, 0)
+        val build = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {

```

Comment:
I don't understand why you need this if statement because I don't have enough familiarity with the code.

Reply:
On newer android versions we need to use `longVersionCode` which is not present on android 7, so we fallback to  `versionCode`

Reply:
fwiw glean_parser can generate this at build time (it uses `BuildConfig.VERSION_CODE` and `BuildConfig.VERSION_NAME`).

If you're using the Glean Gradle plugin it does that for you and you should be able to import it:

```
import org.mozilla.firefox.vpn.daemon.GleanMetrics.GleanBuildInfo
```	

and then pass that in as `buildInfo = GleanBuildInfo.buildInfo`